chain/src/types.rs

```diff
@@ -114,37 +144,54 @@ impl SyncState {
 	}
 
 	/// Update the syncing status
-	pub fn update(&self, new_status: SyncStatus) {
-		if self.status() == new_status {
```

Comment:
This check is not great, we take a read lock, compare and then release the lock, so when we obtain a write lock the state may be different already.